tooling/generators/provider/provider.tmpl.ts

```diff
@@ -0,0 +1,43 @@
+import {Injectable} from 'angular2/core';
+import {Http} from 'angular2/http';
+import 'rxjs/add/operator/map';
+
+/*
+  Generated class for the <%= jsClassName %> provider.
+
+  See https://angular.io/docs/ts/latest/guide/dependency-injection.html
+  for more info on providers and Angular 2 DI.
+*/
+@Injectable()
+export class <%= jsClassName %> {
+  private http;
```

Comment:
I'm not sure what the standard for visibility is, I think I'd default to `public` though as the convention is to prefix private so it's more readable when it's public IMO.

I'd go with :

```
data: any

constructor(public http: Http)
```